src/Core/MapUtil/FHMap.cpp

```diff
@@ -115,6 +115,7 @@ void FHMap::initTiles(const Core::IGameDatabase* database)
 
     if (m_tileMapUpdateRequired) {
         m_tileMap.correctTerrainTypes(dirtTerrain, sandTerrain, waterTerrain);
+        m_tileMap.clearRoads(m_towns);
```

Comment:
I would better do same task before saving road objects after generation. We must not do this here. That will drastically affect map editor later.